packages/toolkit/src/query/core/buildMiddleware/cacheCollection.ts

```diff
@@ -100,6 +101,22 @@ export const buildCacheCollectionHandler: InternalHandlerBuilder = ({
         )
       }
     }
+
+    if (
+      queryThunk.fulfilled.match(action) ||
+      queryThunk.rejected.match(action)
+    ) {
+      const state = mwApi.getState()[reducerPath]
+      const queryCacheKey = action.meta.arg.queryCacheKey
```

Comment:
This block only differs from the above in line 71, so I wonder if those could be consoldiated somehow 🤔

Reply:
My newest commit should reflect these conditionals being turned into one. I am sure there's some TS trick to make the conditional simpler yet again if needed, but I am unsure of what that is off the top of my head.

```ts
 if (unsubscribeQueryResult.match(action) ||
      queryThunk.fulfilled.match(action) ||
      queryThunk.rejected.match(action)
    ) {
      const state = mwApi.getState()[reducerPath]
      let queryCacheKey: QueryCacheKey

      if (unsubscribeQueryResult.match(action)) {
        queryCacheKey = action.payload.queryCacheKey
      } else {
        queryCacheKey = action.meta.arg.queryCacheKey
      }
```